urllib3/util/retry.py

```diff
@@ -205,7 +206,7 @@ def get_backoff_time(self):
         return min(self.BACKOFF_MAX, backoff_value)
 
     def parse_retry_after(self, retry_after):
-        if retry_after.isdigit():
+        if re.match(r"^\s*[0-9]+\s*$", retry_after):
```

Comment:
Actually in my opinion that method should be private and not allow whitespace. It's called only through get_retry_after which gets the Retry-After value from a response, and it is in my opinion the business of the response object to supply header values with nonsignificant whitespace stripped.